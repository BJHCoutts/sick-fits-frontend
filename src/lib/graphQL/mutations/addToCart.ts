import { gql } from "@apollo/client"

export const ADD_TO_CART_MUTATION = gql`
	mutation ADD_TO_CART($productId: ID!) {
		addToCart(productId: $productId) {
			id
		# quantity
		# product {
		# 	id
		# 	name
		# }
		# user {
		# 	id
		# 	name
		# 	email
			# }
		}
	}
`