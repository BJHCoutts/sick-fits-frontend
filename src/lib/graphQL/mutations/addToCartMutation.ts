import { gql } from "@apollo/client"

export const ADD_TO_CART_MUTATION = gql`
	mutation ADD_TO_CART_MUTATION($productId: ID!) {
		addToCart(productId: $productId) {
			id
		}
	}
`