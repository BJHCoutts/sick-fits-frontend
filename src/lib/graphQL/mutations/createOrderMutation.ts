import { gql } from "@apollo/client"

export const CREATE_ORDER_MUTATION = gql`
	mutation CREATE_ORDER_MUTATION($token:String!) {
  checkout (token:$token) {
    id
    charge
    total
    items{
      id
      name
    }
  }
}
`