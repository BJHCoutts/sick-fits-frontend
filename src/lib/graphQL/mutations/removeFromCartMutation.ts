import { gql } from "@apollo/client"

export const REMOVE_FROM_CART_MUTATION = gql`
	mutation REMOVE_FROM_CART_MUTATION ($id:ID!){
		deleteCartItem (id:$id) {
			id
			quantity
			product {
				name
			}
		}
	}
`