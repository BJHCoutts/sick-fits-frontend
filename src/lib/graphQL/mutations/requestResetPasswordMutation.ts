import { gql } from "@apollo/client"

export const REQUEST_RESET_PASSWORD_MUTATION = gql`
	mutation REQUEST_RESET_PASSWORD_MUTATION($email: String!) {
		sendUserPasswordResetLink (email:$email) {
      code
      message
    }
	}
`