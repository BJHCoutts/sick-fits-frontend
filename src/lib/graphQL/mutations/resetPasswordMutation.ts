import { gql } from "@apollo/client"

export const RESET_PASSWORD_MUTATION = gql`
	mutation RESET_PASSWORD_MUTATION(
		$email: String!, 
		$password: String!, 
		$token: String!
	) {
		redeemUserPasswordResetToken(
			email: $email
			password: $password
			token: $token
		) {
			code
			message
		}
	}
`