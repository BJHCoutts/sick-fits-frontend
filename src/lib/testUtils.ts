import casual from 'casual'
import { PAGINATION_QUERY } from './graphQL/queries/paginationQuery'
import { TProduct } from './types/TProduct'
import { TCartItem } from './types/TCartItem'
import { TUser } from './types/TUser'

// seed it so we get consistent results
casual.seed(777)

export const fakeItem = (): TProduct => ({
	__typename: 'Item',
	id: 'abc123',
	price: 5000,
	user: null,
	photo: {
		id: 'abc123',
		altText: 'dogs are best',
		image: {
			id: 'abc123',
			publicUrlTransformed: 'dog.jpg',
		},
	},
	name: 'dogs are best',
	description: 'dogs',
	status: 'AVAILABLE',
})

export const fakeUser = (overrides: {}): TUser => ({
	__typename: 'User',
	id: '4234',
	name: casual.name,
	email: casual.email,
	permissions: ['ADMIN'],
	orders: [],
	cart: [],
	...overrides,
})

// const fakeOrderItem = () => ({
//   __typename: 'OrderItem',
//   id: casual.uuid,
//   image: {
//     image: `${casual.word}.jpg`,
//   },
//   name: casual.words(),
//   price: 4234,
//   quantity: 1,
//   description: casual.words(),
// });

// const fakeOrder = () => ({
//   __typename: 'Order',
//   id: 'ord123',
//   charge: 'ch_123',
//   total: 40000,
//   items: [fakeOrderItem(), fakeOrderItem()],
//   createdAt: '2022-12-11T20:16:13.797Z',
//   user: fakeUser(),
// });

export const fakeCartItem = (overrides: {}): TCartItem => ({
	__typename: 'CartItem',
	id: 'omg123',
	quantity: 4,
	product: fakeItem(),
	user: fakeUser({}),
	...overrides,
})

// // Fake LocalStorage
// class LocalStorageMock {
//   constructor() {
//     this.store = {};
//   }

//   clear() {
//     this.store = {};
//   }

//   getItem(key) {
//     return this.store[key] || null;
//   }

//   setItem(key, value) {
//     this.store[key] = value.toString();
//   }

//   removeItem(key) {
//     delete this.store[key];
//   }
// }

export function makePaginationMocksFor(length: number) {
	return [
		{
			request: { query: PAGINATION_QUERY },
			result: {
				data: {
					_allProductsMeta: {
						count: length,
					},
					itemsConnection: {
						__typename: 'aggregate',
						aggregate: {
							count: length,
							__typename: 'count',
						},
					},
				},
			},
		},
	]
}

// export
// makePaginationMocksFor,
// LocalStorageMock,
// fakeItem,
// fakeUser,
// fakeCartItem,
// fakeOrder,
// fakeOrderItem,
