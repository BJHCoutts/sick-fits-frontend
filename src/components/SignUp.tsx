import { useMutation } from '@apollo/client'
import useForm from '../lib/functions/useForm'
import SForm from './styles/SForm'
import { SIGN_IN_MUTATION } from '../lib/graphQL/mutations/signInMutation'
import { CURRENT_USER_QUERY } from '../lib/graphQL/queries/currentUserQuery'

export default function SignUp() {
	const { inputs, handleChange, resetForm } = useForm({
		email: '',
		password: '',
	})

	const [signIn, { error, loading }] = useMutation(SIGN_IN_MUTATION, {
		variables: inputs,
		refetchQueries: [{ query: CURRENT_USER_QUERY }],
	})

	async function handleSubmit(e: React.FormEvent) {
		e.preventDefault()
		const res = await signIn()
		resetForm()
	}

	return (
		<>
			<SForm method="POST" onSubmit={handleSubmit}>
				<h2>Sign Up for Your Account</h2>
				<fieldset>
					<label htmlFor="email">
						Email
						<input
							type="email"
							name="email"
							id="email"
							placeholder="Your Email Address"
							autoComplete="email"
							value={inputs.email}
							onChange={handleChange}
						/>
					</label>
					<label htmlFor="name">
						Name
						<input
							type="name"
							name="name"
							id="name"
							placeholder="Your Name"
							autoComplete="name"
							value={inputs.name}
							onChange={handleChange}
						/>
					</label>
					<label htmlFor="password">
						Password
						<input
							type="password"
							name="password"
							id="password"
							placeholder="Your Password"
							autoComplete="password"
							value={inputs.password}
							onChange={handleChange}
						/>
					</label>
					<button type="submit">Sign In</button>
				</fieldset>
			</SForm>
		</>
	)
}
