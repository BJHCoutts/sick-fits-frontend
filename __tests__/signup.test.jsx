import { fakeUser } from '../src/lib/testUtils'
import { SIGN_UP_MUTATION } from '../src/lib/graphQL/mutations/signUpMutation'
// import { CURRENT_USER_QUERY } from '../src/lib/graphQL/queries/currentUserQuery'
import { MockedProvider } from '@apollo/client/testing'
import { render, screen } from '@testing-library/react'
import SignUp from '../src/components/SignUp'
import userEvent from '@testing-library/user-event'

const user = fakeUser()
const password = 'password'

const mocks = [
	{
		request: {
			query: SIGN_UP_MUTATION,
			variables: {
				name: user.name,
				email: user.email,
				password,
			},
		},
		result: {
			data: {
				createUser: {
					__typename: 'User',
					id: 'abc123',
					name: user.name,
					email: user.email,
				},
			},
		},
	},
	// {
	// 	request: { query: CURRENT_USER_QUERY },
	// 	result: { data: { authenticatedItem: user } },
	// },
]

describe('<SignUp />', () => {
	it('renders and matches the snapshot', async () => {
		const { container, debug } = render(
			<MockedProvider mocks={mocks}>
				<SignUp />
			</MockedProvider>
		)
		expect(container).toMatchSnapshot()

		await userEvent.type(screen.getByPlaceholderText('Your Name'), user.name)

		await userEvent.type(screen.getByPlaceholderText('Your Email'), user.email)

		await userEvent.type(screen.getByPlaceholderText('Your Password'), password)

		await userEvent.click(screen.getByText('Sign Up'))

		await screen.findByText(`Signed Up with ${user.email}`)
	})
})
