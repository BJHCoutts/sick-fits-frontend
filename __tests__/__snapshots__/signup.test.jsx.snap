// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUp /> renders and matches the snapshot 1`] = `
<div>
  <form
    class="sc-bdfCDU cdIIGl"
    method="POST"
  >
    <h2>
      Sign Up for Your Account
    </h2>
    <fieldset
      aria-busy="false"
    >
      <label
        for="email"
      >
        Email
        <input
          autocomplete="email"
          name="email"
          placeholder="Your Email"
          type="email"
          value=""
        />
      </label>
      <label
        for="name"
      >
        Name
        <input
          autocomplete="name"
          id="name"
          name="name"
          placeholder="Your Name"
          type="name"
          value=""
        />
      </label>
      <label
        for="password"
      >
        Password
        <input
          autocomplete="password"
          data-testid="password"
          id="password"
          name="password"
          placeholder="Your Password"
          type="password"
          value=""
        />
      </label>
      <button
        type="submit"
      >
        Sign Up
      </button>
    </fieldset>
  </form>
</div>
`;
